frontends/p4/sideEffects.cpp

```diff
@@ -430,6 +595,10 @@ class DismantleExpression : public Transform {
                         break;
                     if (!p1->hasOut() && !p2->hasOut())
                         continue;
+                    if (useTemporary.find(p1) != useTemporary.end())
```

Comment:
idiomatically, I prefer `!useTemporary.count(p1)` to test if something is not in a set or map.  Mostly because it is easier to apply to something that has side effects (such as the result of a call that returns a set)